File: lib/models/user_model.dart
class UserModel {
  final int id;
  final String name;
  final String email;
  final String role;
  final String token;

  UserModel({
    required this.id,
    required this.name,
    required this.email,
    required this.role,
    this.token = '',
  });
}

File: backend/src/models/user.model.js
const db = require('../config/db')

exports.create = (data, callback) => {
  db.run(
    `INSERT INTO users (name, email, password, role)
     VALUES (?,?,?,?)`,
    [data.name, data.email, data.password, data.role],
    callback
  )
}

exports.findByEmail = (email, callback) => {
  db.get(
    `SELECT * FROM users WHERE email = ?`,
    [email],
    callback
  )
}
